### ISSUE TITLE
No artifact published under `silencer-lib_2.12` and `silencer-lib_2.13` for 1.4.3

### ISSUE THREAD

**Issue by mihaisoloi** [ghik/silencer#37]
There are artifacts published for individual versions of `silencer` for 2.13.0 and 2.12.10 but that requires adding them like this in SBT:

```
libraryDependencies ++= Seq(
  compilerPlugin("com.github.ghik" % "silencer-plugin_2.12.10" % silencerVersion cross CrossVersion.full),
  "com.github.ghik" % "silencer-lib_2.12.10" % silencerVersion % Provided cross CrossVersion.full
)
```
whereas if they would be published under the correct artifactId we could use the `%%` notation

This was fine for `1.4.2` but not so for `1.4.3`. Example:
https://mvnrepository.com/artifact/com.github.ghik/silencer-lib_2.13/1.4.2
https://mvnrepository.com/artifact/com.github.ghik/silencer-lib_2.13.0/1.4.3

**Comment by raboof** [comment]
For the `compilerPlugin`, this is actually intended, since the compiler API is not binary compatible across compiler releases. It can be written a little nicer:

```
compilerPlugin("com.github.ghik" %% "silencer-plugin" % silencerVersion cross CrossVersion.full),
"com.github.ghik" %% "silencer-lib" % silencerVersion % Provided cross CrossVersion.full))
```

For `silencer-lib` I guess it's debatable, but perhaps neat to keep it consistent?

**Comment by ghik** [comment]
Yes, as explained in #31 artifacts must use full Scala version. As @raboof mentioned, `silencer-lib` technically could be exempt from that but IMO it would be ugly having to declare `silencer-lib` dependency differently from `silencer-plugin` and it would provide no value.

**Comment by mihaisoloi** [comment]
makes sense, thank you for the explanation
